Don't explicitly handle z_eof in gzread(), fall through and let the lack
of any available data give us our return value.  This avoids triggering
an infinite loop in man-db - I'm not able to convince myself that this
isn't an issue in man-db at the minute but this avoids the issue.

Index: debian/gzio.c
===================================================================
--- debian.orig/gzio.c	2009-12-25 15:52:36.000000000 +0000
+++ debian/gzio.c	2009-12-25 15:52:43.000000000 +0000
@@ -432,7 +432,7 @@
     if (s == NULL || s->mode != 'r') return Z_STREAM_ERROR;
 
     if (s->z_err == Z_DATA_ERROR || s->z_err == Z_ERRNO) return -1;
-    if (s->z_err == Z_STREAM_END || s->z_eof) return 0;  /* EOF */
+    if (s->z_err == Z_STREAM_END) return 0;  /* EOF */
 
     next_out = (Byte*)buf;
     s->stream.next_out = (Bytef*)buf;
